src/backend/distributed/metadata/node_metadata.c

```diff
@@ -2871,15 +2871,15 @@ TupleToWorkerNode(TupleDesc tupleDescriptor, HeapTuple heapTuple)
 	 */
 	heap_deform_tuple(heapTuple, tupleDescriptor, datumArray, isNullArray);
 
-	char *nodeName = DatumGetCString(datumArray[Anum_pg_dist_node_nodename - 1]);
-	char *nodeRack = DatumGetCString(datumArray[Anum_pg_dist_node_noderack - 1]);
+	char *nodeName = TextDatumGetCString(datumArray[Anum_pg_dist_node_nodename - 1]);
```

Comment:
It seems we have more `DatumGetCString` in the code. Can you double check if we really use them on "cstring" or do we have any cases like this?

Reply:
It seems that `DatumGetCString` returns a unicode string. I see such errors if I use it:
```SQL
SELECT FROM master_add_node('localhost', 57637);
ERROR:  could not find valid entry for node "^Ulocalhost:57637"
```

Reply:
iirc, text is prefixed with the length, whereas cstring is not, so if you try to use a cstring as text it sees some weird initial character.